kafka/cluster.go

```diff
@@ -89,11 +91,12 @@ func NewCluster(opts *options.Options, storageCh chan<- *StorageRequest) *Cluste
 	connectionLogger.Info("successfully connected to kafka cluster")
 
 	return &Cluster{
-		storageCh: storageCh,
-		client:    client,
-		admin:     admin,
-		logger:    logger,
-		options:   opts,
+		storageCh:   storageCh,
+		client:      client,
+		admin:       admin,
+		logger:      logger,
+		options:     opts,
+		topicFilter: regexp.MustCompile(opts.TopicFilter),
```

Comment:
This panics. I think we should use `Compile` instead, check for an error and print a proper error message with os.Exit(1) afterwards. I think I used logrus here, maybe it has something similar like the zap's "fatal" level.

Rationale: Panic logs might cause trouble with Logstash / ELK / logging pipelines so I want to avoid them where possible.